File: src/classes/api/req_res/add-ur-log-response.ts
'use strict'

import { URLog } from '@/classes/datas/ur-log'
import { GkillAPIResponse } from '../gkill-api-response'
import { Kyou } from '@/classes/datas/kyou'

export class AddURLogResponse extends GkillAPIResponse {

    added_urlog: URLog

    added_kyou: Kyou | null

    constructor() {
        super()
        this.added_urlog = new URLog()
        this.added_kyou = null
    }

}

File: src/server/gkill/api/req_res/add_ur_log_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/server/gkill/api/message"
	"github.com/mt3hr/gkill/src/server/gkill/dao/reps"
)

type AddURLogResponse struct {
	Messages message.GkillMessages `json:"messages"`

	Errors message.GkillErrors `json:"errors"`

	AddedURLog *reps.URLog `json:"added_urlog"`

	AddedKyou *reps.Kyou `json:"added_kyou"`
}
